### DISCUSSION TITLE
Top level await

### DISCUSSION THREAD

**Discussion by ianwalter** [eslint/eslint#13934] [General]
Can someone re-open this issue? https://github.com/eslint/eslint/issues/13178

Top-level await is supported in Node v15 but I'm getting this error when using them:

```
5:13  error  Parsing error: Cannot use keyword 'await' outside an async function
```

As an aside, is there a way to disable parsing errors?

**Comment by anikethsaha** [comment]
top level await is still in stage 3. eslint supports only stage 4  proposals.
IMO, lets wait for proposal to reach stage 4

**Reply by ianwalter** [reply]
The stage is only relevant in the browser environment though. Node.js is its own environment. So if ESLint supports Node, it should support parsing TLA.

Looks like TLA was actually unflagged in Node v14.8 and v14 is the current Long Term Support version so this feature is not experimental or going away.

**Reply by g-plane** [reply]
The "Stage" concept is from ECMAScript, not about any environment or runtime. ESLint just runs on Node.js, but it can check any JavaScript code which uses Stage 4 syntaxes.

**Reply by ianwalter** [reply]
Ok, fair, I guess ESLint has the line drawn there. Thank you all for your responses.

**Comment by aladdin-add** [comment]
to support stage<4 features, please use `babel-eslint`!

**Comment by ghost** [comment]
Ok in case anyone stumbles down the rabbit hole of "non-babel stage 3 espree", I believe this is working:

in your `eslintrc`:

```
"parser": "./path/to/a/new/file.js"
```

Then in this file, inject whatever extensions you want and expose as an eslint parser (e.g. by copying the contents of `eslint.js` and modifying slightly

```javascript
"use strict";

const acorn = require("acorn");
const stage3 = require("acorn-stage3");
const espree = require("espree/lib/espree");

// To initialize lazily.
const parsers = {
  _regular: null,

  get regular() {
    if (this._regular === null) {
      //  here
      this._regular = acorn.Parser.extend(espree()).extend(stage3);
    }
    return this._regular;
  },

  get(options) {
    return this.regular;
  },
};

function tokenize(code, options) {
  const Parser = parsers.get(options);

  // Ensure to collect tokens.
  if (!options || options.tokens !== true) {
    options = Object.assign({}, options, {tokens: true}); // eslint-disable-line no-param-reassign
  }

  return new Parser(options, code).tokenize();
}

function parse(code, options) {
  const Parser = parsers.get(options);

  return new Parser(options, code).parse();
}
exports.parseForESLint = function (code, options) {
  return {
    ast: parse(code, options),
    services: null,
    scopeManager: null,
    visitorKeys: null,
  };
};

```

**Comment by mechanoid** [comment]
[top level await is Stage 4 now](https://github.com/tc39/proposal-top-level-await) IMHO https://github.com/eslint/eslint/issues/13178 should be opened up again.

**Reply by nimit2801** [reply]
#14632 also support this conversation and any updates there would really help people finding a solution for the top-level await 😄

**Comment by hholst80** [comment]
It's been a year. Is this project relevant going forward or have everyone moved on to something else?

**Comment by aladdin-add** [comment]
TLA has been supported in eslint v8(ecmaVersion:2022).
